### PR TITLE
Parse authorized user

### PR DESCRIPTION
**Pull request by nk-coding**
https://frontend.gropius.duckdns.org/components/1c93546b-5144-403f-9026-6fe82804b1b3/issues/7b71a954-a35b-446c-8dac-72df20b5575d

- [ ] Requirements of the issue are met
- [ ] Code has been reviewed
- [ ] Code is documented
- [ ] GraphQL related artefacts are documented

### CODE REVIEW

**Review by lesser1 — APPROVED** [review]

**Comment by lesser1** [comment]
I suggest that we get another approval since I do not know too much about this stuff and I could be that I did not see any potential issues.

File: envelopPlugins.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import { DefaultContext } from "@envelop/core";
+import { useGenericAuth } from "@envelop/generic-auth";
+import * as jose from "jose";
+
+type UserType =
+  | {
+      id: string;
+      roles: string[];
+    }
+  | {};
```
Comment (delvh) [inline envelopPlugins.ts:10]: That is valid TS?

File: envelopPlugins.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    const authHeader = (context.request as any)?.headers?.get("authorization");
+    if (authHeader == undefined) {
+      return {};
+    }
+    let parsedHeader: string;
+    if (authHeader.startsWith("Bearer ")) {
+      parsedHeader = authHeader.substring(7);
+    } else {
+      parsedHeader = authHeader;
+    }
```
Comment (delvh) [inline envelopPlugins.ts]: Java!
Where's your goddamn TrimPrefix!

File: envelopPlugins.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  context: DefaultContext
+): Promise<UserType | null> {
+  try {
+    const authHeader = (context.request as any)?.headers?.get("authorization");
+    if (authHeader == undefined) {
+      return {};
+    }
+    let parsedHeader: string;
+    if (authHeader.startsWith("Bearer ")) {
+      parsedHeader = authHeader.substring(7);
```
Comment (delvh) [inline envelopPlugins.ts]:
```suggestion
      parsedHeader = authHeader.substring("Bearer ".length());
```

File: envelopPlugins.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    } else {
+      parsedHeader = authHeader;
```
Comment (delvh) [inline envelopPlugins.ts]: Isn't that an error case here?
Should we really accept password based Auth?

File: envelopPlugins.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    }
+    const { payload } = await jose.jwtVerify(parsedHeader, JWKS, {});
+    const user: UserType = {
+      id: payload.sub!,
+      roles: getValidRoles(payload),
+    };
+    context.authHeader = parsedHeader;
```
Comment (delvh) [inline envelopPlugins.ts]: Wait, without the "Bearer"?
Why?
This sounds like a bad idea for the sake of removing a few lines.

File: envelopPlugins.ts (hunk 1 above)
Comment (nk-coding) [reply envelopPlugins.ts:10]: sure

File: envelopPlugins.ts (hunk 4 above)
Comment (nk-coding) [reply envelopPlugins.ts]: this is still the same JWT-based auth
this just allows users to be a bit lazy and omit the "Bearer ", which is in particular helpful when debugging stuff via the Gateway

Comment (nk-coding) [reply envelopPlugins.ts]: changed with 7c70647811b735464e80c89c6947bf49fff5d0c5

File: envelopPlugins.ts (hunk 5 above)
Comment (nk-coding) [reply envelopPlugins.ts]: changed with 7c70647811b735464e80c89c6947bf49fff5d0c5

**Review by delvh — APPROVED** [review]

**Pull request merged**